### PR TITLE
TSFile - remove from background_fetch plugin.

### PR DESCRIPTION
**Pull request by SolidWallOfCode**
`TSFile` and related plugin API are being deprecated. This removes those from the `background_fetch` plugin, along with some other clean up.

### CODE REVIEW

**Comment by SolidWallOfCode** [comment]
https://github.com/apache/trafficserver/blob/master/tests/gold_tests/cache/background_fill.test.py

**Comment by SolidWallOfCode** [comment]
Hmmm, I'll have to double check.

File: plugins/background_fetch/configs.h
Comment (ezelkow1) [inline plugins/background_fetch/configs.h:1]: that extern DbgCtl Bg_dbg_ctl should probably be

DbgCtl Bg_dbg_ctl{PLUGIN_NAME};

Plugin will build as-is but not run like that due to the externed dbg

**Review by ezelkow1 — CHANGES_REQUESTED** [review]

File: plugins/background_fetch/configs.h
Comment (ywkaras) [reply plugins/background_fetch/configs.h:1]: Could just add:

DbgCtl Bg_dbg_ctl{PLUGIN_NAME};

to config.cc .

**Comment by bryancall** [comment]
@ezelkow1 is going to take over this PR

**Comment by bryancall** [comment]
This was fixed with #10980

**Pull request closed without merging**